functionary/train/custom_datasets.py

```diff
@@ -576,3 +701,10 @@ def stat(self):
         print(
             f"original avg length: {original_avg_length}; avg packed length: {avg_packed_length}"
         )
+
+
+class FAPackedDataset(PackedDataset):
```

Comment:
Maybe you can set an attribute `use_flash_attention_2` in `PackedDataset` instead of creating new subclass? Reduces the complication